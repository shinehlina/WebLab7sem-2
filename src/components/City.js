import React from 'react';

class City extends React.Component {
  render() {
    return (
      <div>
        <h2> {this.props.cityName} </h2>
        <button onClick={() => this.props.deleteCity(this.props.index)}>Delete</button>
        <p> Coordinates: <i>[{this.props.longtitude} : {this.props.latitude}]</i></p>
        <p> Weather description: <i>{this.props.description}</i></p>
        <p> Temperature: <i>{this.props.temperature}K </i></p>
        <p> Humidity: <i>{this.props.humidity}%</i></p>
        <p> Wind: <i>{this.props.windSpeed}m/c</i></p>
      </div>
    );
  }
}

export default City;
