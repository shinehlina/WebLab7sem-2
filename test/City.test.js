import React from "react";
import City from "../src/components/City";
import { shallow } from './enzyme';
import { shallowToJson } from "enzyme-to-json";

it("City didn't changed", () => {
  const app = shallow(
    <City
      latitude={30}
      longtitude={60}
      description={"windy"}
      temperature={273}
      humidity={20}
      windSpeed={10}
      pressure={1000}
      cityName={"Moscow"}
      icon={1}
      index={0}
    />
  );
  expect(shallowToJson(app)).toMatchSnapshot();
});
