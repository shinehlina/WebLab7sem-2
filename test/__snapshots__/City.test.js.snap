// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City didn't changed 1`] = `
<div>
  <Col
    className="valign-wrapper"
    m={3}
    s={3}
  >
    <h6>
       
      Moscow
       
    </h6>
  </Col>
  <Col
    className="valign-wrapper"
    m={3}
    s={3}
  >
    <h6
      style={
        Object {
          "float": "right",
          "fontWeight": "bold",
        }
      }
    >
      0
      หะก
    </h6>
  </Col>
  <Col
    className="valign-wrapper"
    m={3}
    s={4}
  >
    <img
      alt="icon"
      src="https://openweathermap.org/img/wn/1.png"
    />
  </Col>
  <Col
    className="valign-wrapper"
    m={3}
    s={3}
  >
    <Button
      className="grey darken-1"
      floating={true}
      icon={
        <Icon>
           remove 
        </Icon>
      }
      node="button"
      onClick={[Function]}
      small={true}
      waves="light"
    />
  </Col>
  <CityDescription
    description="windy"
    humidity={20}
    latitude={30}
    longtitude={60}
    pressure={1000}
    temperature={273}
    windSpeed={10}
  />
</div>
`;
